src/main/java/net/snowflake/ingest/streaming/internal/AbstractRowBuffer.java

```diff
@@ -543,10 +556,32 @@ static <T> AbstractRowBuffer<T> createRowBuffer(
                 channelRuntimeState,
                 enableParquetMemoryOptimization,
                 maxChunkSizeInBytes,
-                maxRowSizeInBytes);
+                maxRowSizeInBytes,
+                insertRowsRecommendedMaxSizeInBytes,
+                insertRowsEnforcedMaxSizeInBytes);
       default:
         throw new SFException(
             ErrorCode.INTERNAL_ERROR, "Unsupported BDEC format version: " + bdecVersion);
     }
   }
+
+  private void checkBatchSizeEnforcedMaximum(float batchSizeInBytes) {
+    if (batchSizeInBytes > insertRowsEnforcedMaxSizeInBytes) {
+      throw new SFException(
+          ErrorCode.MAX_BATCH_SIZE_EXCEEDED,
+          insertRowsEnforcedMaxSizeInBytes,
+          insertRowsRecommendedMaxSizeInBytes);
+    }
+  }
+
+  private void checkBatchSizeRecommendedMaximum(float batchSizeInBytes) {
+    if (batchSizeInBytes > insertRowsRecommendedMaxSizeInBytes) {
+      logger.logWarn(
+          "Too large batch of rows passed to 'insertRows': {} bytes. The recommended max batch size"
```

Comment:
nit: since this is just a recommended error warning, can we make this message less strict? 
"Too large batch of rows passed to 'insertRows': {} bytes. The recommended max batch is {} bytes" -> "The batch of rows passed to 'insertRows' is over the recommended max batch size. Given {} bytes, recommended max batch is {} bytes."

Reply:
Good point, I reworded it like this, what do you think?

Warning:
> The batch of rows passed to 'insertRows' is over the recommended max batch size. Given 8.3886112E7 bytes, recommended max batch is 16777216 bytes. For optimal performance and memory utilization, we recommend splitting large batches into multiple smaller ones and call insertRows for each smaller batch separately.

Error:
> Too large batch of rows passed to insertRows, the batch size cannot exceed 128,000,000 bytes, recommended batch size for optimal performance and memory utilization is 16,777,216 bytes. We recommend splitting large batches into multiple smaller ones and call insertRows for each smaller batch separately.

Reply:
Would it be possible to also include the given batch size in the Error message?

Too large batch of rows passed to insertRows, **given xxx bytes.** The batch size cannot exceed 128,000,000 bytes...

Otherwise looks good!

Reply:
We don't know the full batch size because we stop processing once we exceed the limit.